src/App.tsx

```diff
@@ -1,61 +1,144 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { ToDo } from './types/type';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(id: number) {
+  return usersFromServer.find(usr => (id) === usr.id) || null;
+}
 
 export const App = () => {
+  const initTodos = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUser(todo.userId),
+  }));
+
+  const [todos, setTodos] = useState<ToDo[]>(initTodos);
+  const [todo, setTodo] = useState<ToDo>(
+    {
+      id: 0,
+      title: '',
+      completed: false,
+      user: null,
+    },
+  );
+
+  const [isFilled, setIsFilled] = useState(true);
+
+  const onAdd = (newTodo: ToDo) => {
+    setTodos(currentTodos => [
+      ...currentTodos,
+      {
+        ...newTodo,
+        id: Math.max(...(todosFromServer.map(toDo => toDo.id))) + 1,
+      },
+    ]);
+  };
+
+  const resetForm = () => {
+    setTodo(
+      {
+        id: 0,
+        title: '',
+        completed: false,
+        user: null,
+      },
+    );
+
+    setIsFilled(true);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTodo(
+      { ...todo, title: event.target.value },
+    );
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setTodo(
+      {
+        ...todo,
+        user: getUser(+event.target.value),
+      },
+    );
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todo.title || !todo.user?.id) {
+      setIsFilled(false);
+
+      return;
+    }
+
+    onAdd(todo);
+    resetForm();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
```

Comment:
You can even move this whole form logic into a separate component, like TodoForm.tsx, and use it here as:

`<TodoList todos={todos} />`